### ISSUE TITLE
Undefined error: 0

### ISSUE THREAD

**Issue by fj0n** [AlexeyAB/darknet#4211]
[yolov3-tiny_drinks.txt](https://github.com/AlexeyAB/darknet/files/3801611/yolov3-tiny_drinks.txt)

Hi there,

I'm for the first time trying to train yolov3 for my custom objects. I have 3 classes that I want to detect and prepared everything according to the instructions here.

But when I try to train the model with:
```./darknet detector train cfg/obj.data cfg/yolov3-tiny_drinks.cfg darknet53.conv.74```

the following error is prompted:
```
 20 route  19 8
   21 Layer before convolutional layer must output image.: Undefined error: 0
Assertion failed: (0), function error, file ./src/utils.c, line 256.
Abort trap: 6
```
Can someone direct me in the right path?

Thanks upfront!

**Comment by AlexeyAB** [comment]
Show your cfg-file

**Comment by fj0n** [comment]
Sure, here it is:

[net]
# Testing
batch=64
subdivisions=8
# Training
# batch=64
# subdivisions=8
width=400
height=400
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 6000
policy=steps
steps=4800,5400
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=24
activation=linear



[yolo]
mask = 3,4,5
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=3
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=24
activation=linear

[yolo]
mask = 0,1,2
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=3
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

**Comment by fj0n** [comment]
> Show your cfg-file

I uploaded it!

Turned out, that when I change the resolution back to 416x416, it starts to work. But it doesn't make sense to me, because my images have a resolution von 960x688.

**Comment by AlexeyAB** [comment]
> 
> width=400
> height=400

https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> increase network resolution in your .cfg-file (height=608, width=608 or any value multiple of 32) - it will increase precision

**Comment by fj0n** [comment]
@AlexeyAB you are right! Thanks!
